src/backend/distributed/executor/executor.c

```diff
@@ -600,6 +604,12 @@ FinishDistributedExecution(DistributedExecution *execution)
 		/* prevent copying shards in same transaction */
 		XactModificationLevel = XACT_MODIFICATION_DATA;
 	}
+
+	if (execution->waitEventSet != NULL)
+	{
+		FreeWaitEventSet(execution->waitEventSet);
```

Comment:
Would it be sufficient to do this at the end of `RunDistributedExecution`. If so, can `waitEventSet` remain a local variable?

Reply:
done